features/src/main/scala/com/salesforce/op/stages/impl/feature/TimePeriod.scala

```diff
@@ -29,28 +29,33 @@
  */
 package com.salesforce.op.stages.impl.feature
 
+import java.time.temporal.WeekFields
+import java.time.{Instant, LocalDateTime, ZoneId}
+
 import com.salesforce.op.utils.date.DateTimeUtils
 import enumeratum.{Enum, EnumEntry}
-import org.joda.time.{DateTime => JDateTime}
 
+case class TimePeriodVal(value: Int, min: Int, max: Int)
+
+sealed abstract class TimePeriod(extractFn: LocalDateTime => TimePeriodVal) extends EnumEntry with Serializable {
+  def extractTimePeriodVal: Long => TimePeriodVal =
```

Comment:
@gerashegalov  please make `extractTimePeriodVal` and `extractFn` regular functions to avoid extra memory allocations